components/micro-gateway-core/src/main/ballerina/src/gateway/handlers/jwt_auth_handler.bal

```diff
@@ -237,46 +233,15 @@ public function generateAndSetBackendJwtHeader(string credential,
                 printDebug(KEY_JWT_AUTH_PROVIDER, "decoded token credential");
                 // get the subscribedAPI details
                 map<string> apiDetails = getAPIDetails(payload, apiName, apiVersion);
-                // checking if cache is enabled
-                if (enabledCaching) {
-                    var cachedToken = jwtGeneratorCache.get(cacheKey);
-                    printDebug(KEY_JWT_AUTH_PROVIDER, "Key: " + cacheKey);
-                    if (cachedToken is string) {
-                        printDebug(KEY_JWT_AUTH_PROVIDER, "Found in jwt generator cache");
-                        printDebug(KEY_JWT_AUTH_PROVIDER, "Token: " + cachedToken);
-                        (jwt:JwtPayload | error) cachedPayload = getDecodedJWTPayload(cachedToken, issuer);
-                        if (cachedPayload is jwt:JwtPayload) {
-                            int currentTime = getCurrentTime();
-                            int? cachedTokenExpiry = cachedPayload?.exp;
-                            if (cachedTokenExpiry is int) {
-                                cachedTokenExpiry = cachedTokenExpiry * 1000;
-                                int difference = (cachedTokenExpiry - currentTime);
-                                if (difference < skewTime) {
-                                    printDebug(KEY_JWT_AUTH_PROVIDER, "JWT regenerated because of the skew time");
-                                    status = setJWTHeader(<@untainted>payload, req, cacheKey, enabledCaching, apiDetails,
-                                                            remoteUserClaimRetrievalEnabled);
-                                } else {
-                                    req.setHeader(jwtheaderName, cachedToken);
-                                    status = true;
-                                }
-                            } else {
-                                printDebug(KEY_JWT_AUTH_PROVIDER, "Failed to read exp from cached token");
-                                return false;
-                            }
-                        }
-                    } else {
-                        printDebug(KEY_JWT_AUTH_PROVIDER, "Could not find in the jwt generator cache");
-                        status = setJWTHeader(<@untainted>payload, req, cacheKey, enabledCaching, apiDetails,
-                                            remoteUserClaimRetrievalEnabled);
-                    }
-                } else {
-                    printDebug(KEY_JWT_AUTH_PROVIDER, "JWT generator caching is disabled");
-                    status = setJWTHeader(<@untainted>payload, req, cacheKey, enabledCaching, apiDetails,
-                                            remoteUserClaimRetrievalEnabled);
-                }
+                BackendJWTGenUserContextDTO tokenContextDTO = {
+                    issuer : issuer,
+                    remoteUserClaimRetrievalEnabled : remoteUserClaimRetrievalEnabled,
+                    payload : payload
```

Comment:
this payload should be retrieved from "invocationContext[PRINCIPAL][ISSUER_CLAIMS]" to support claim mapping. Change in the line 231. See issue - https://github.com/wso2/product-microgateway/issues/1342#issuecomment-662006232

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/1345/commits/cff53e2f64a348347aaa877d1abc7e7c20bfd9d8